storage/tianmu/core/engine_execute.cpp

```diff
@@ -381,6 +378,8 @@ int Engine::Execute(THD *thd, LEX *lex, Query_result *result_output, Query_expre
   Query query(current_txn_);
   CompiledQuery cqu;
 
+  if (result_output->start_execution(thd)) return RETURN_QUERY_TO_MYSQL_ROUTE;
```

Comment:
F.I.Y: `JOIN::prepare_result()`

Reply:
By analyzing MySQL and tianmu engine code, find that JOIN:: prepare_result() function is used for explain query,used in explain_query_specification() function,.

Reply:
get it

Reply:
Pls, Making sure that which type of Query_result it is. Because `select`, `insert`, `delete`, and etc., do different works in their `Query_result::start_execution`.

Reply:
ok，I will do it